### PR TITLE
chore/use virtualenv for docker python

### PR DESCRIPTION
**Pull request by dabeeeenster**
Based on https://pythonspeed.com/articles/activate-virtualenv-dockerfile/

### CODE REVIEW

File: api/docker/docker [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 set -e
 
-python manage.py migrate
+python src/manage.py migrate
```
Comment (dabeeeenster) [inline api/docker/docker]: Bit suspicious of why I need to do this?

File: api/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,12 @@
 FROM python:3.8-slim as application
 
 WORKDIR /app
+
+# Install python packages locally in a virtualenv
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```
[the remark this answers is not in the document]
Comment (dabeeeenster) [reply api/Dockerfile:8]: Surely line 7 removes the need for that? This code works and the docker image runs ok?

File: api/docker/docker (hunk 1 above)
Comment (dabeeeenster) [reply api/docker/docker]: I'm ok with it staying in /src I don't see any need to move that?

File: api/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewelwell) [reply api/Dockerfile:8]: I don't know for sure but I don't _think_ that's correct. 

Are you sure that the subsequent pip install is using the correct pip installation in the virtualenv? And then when you run the application - is it using the correct python binary?

File: api/docker/docker (hunk 1 above)
Comment (matthewelwell) [reply api/docker/docker]: Other than aesthetics, no, not really. It's not important, just feel like we should do it as `src` to me, gives the impression of code in a development environment, not being run in a production container.

File: api/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewelwell) [reply api/Dockerfile:8]: Easiest way to check I guess would be to do something like: 

```
docker build -f api/Dockerfile -t flagsmith-api .
docker run -it flagsmith-api /bin/bash
which python
which pip
pip freeze
```

File: api/docker/docker (hunk 1 above)
Comment (dabeeeenster) [reply api/docker/docker]: What directory name would be better?

File: api/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (dabeeeenster) [reply api/Dockerfile:8]: Looks like its working:

```
nobody@baf2af4cf466:/app$ which python
/opt/venv/bin/python
nobody@baf2af4cf466:/app$ which pip
/opt/venv/bin/pip
```

File: api/docker/docker (hunk 1 above)
Comment (matthewelwell) [reply api/docker/docker]: It should just sit directly in the `app` directory imo.

File: api/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewelwell) [reply api/Dockerfile:8]: Ok 👍

File: api/Dockerfile [hunk 3]
```diff
@@ -17,4 +22,4 @@ EXPOSE 8000
 
 USER nobody
 
-CMD ["./bin/docker"]
+CMD ["./docker/docker"]
```
Comment (matthewelwell) [inline api/Dockerfile]: Should the names change here as well perhaps? Seems a bit weird. Maybe `./scripts/run-docker` or something?

Comment (dabeeeenster) [reply api/Dockerfile]: Seems like an overcomplication to me?

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**